src/main/java/com/betadevels/onlineshopping/action/cart/AddProductToCartAction.java

```diff
@@ -64,6 +64,15 @@ public AddProductToCartResponse invoke()
                 throw new BadRequestException("Requested quantity exceeds availability");
             }
 
+            List<Cart> cartItems=cartHibernateDAO.getCartItemsOfCustomer(customer);
```

Comment:
Do we absolutely have to fetch all the cart items of the customer for checking if the product already exists in the cart?
The database already handles the composite unique key to check this. So we can just avoid a call to the database by looking for db exceptions when inserting a new cart record cartHibernateDAO.create(cart);